portfolio/src/main/webapp/script.js

```diff
@@ -39,24 +40,164 @@ function refreshComments() {
   }
 }
 
+/** Obtain comment data from datastore and display them on page */
 function getData() {
   refreshComments();
-  var numContactsToDisplay = document.getElementById('quantity').value;
-  var link = '/data?numContactsToDisplay=' + numContactsToDisplay
+  const numContactsToDisplay = document.getElementById('quantity').value;
+  const link = '/data?numContactsToDisplay=' + numContactsToDisplay;
 
-  fetch(link).then(response => response.json()).then((contacts) => {
+  // fetch  the data and create a list containing the message of each contact
+  fetch(link).then((response) => response.json()).then((contacts) => {
+    console.log('Hello World');
+    const infoContainer = document.getElementById('more-info');
     contacts.forEach((contact) => {
-      var currContact = document.createElement('li');
-      var userInfo = document.createTextNode(
+      const currContact = document.createElement('li');
+      const userInfo = document.createTextNode(
           contact.firstName + ' ' + contact.lastName + ' says ' +
           contact.subject);
       currContact.appendChild(userInfo);
-      document.getElementById('more-info').appendChild(currContact);
+      infoContainer.appendChild(currContact);
     });
   });
 }
 
 function deleteData() {
-  const request = new Request('/delete-data', {method: 'POST'});
-  fetch(request).then((results) => getData());
+  // upon request navigate to the delete servlet and remove all comments from
+  // datastore
+  if (confirm('Do you wish to Delete all the comments ?')) {
+    const request = new Request('/delete-data', {method: 'POST'});
+    fetch(request).then((results) => getData());
+  }
+}
+
+/** Allow the user to login, when logged in -> display contact form */
+function userLogin() {
+  fetch('/login').then((response) => response.json()).then((currUser) => {
+    document.getElementById('form-overlay-text').innerHTML =
+        currUser.loginMessage;
+    if (currUser.isLoggedIn) {
+      document.getElementById('form-overlay').style.display = 'block';
+      document.getElementById('delete-btn').style.display = 'inline';
+    } else {
+      document.getElementById('form-overlay').style.display = 'none';
+      document.getElementById('delete-btn').style.display = 'none';
+    }
+  });
+}
+
+/** Initialize a map utlizing google's Map API for web */
+function initMap() {
+  // colors for styles
+  const ORANGE = '#d59563';
+  const OPAQUE_BLUE = '#263c3f';
+  const DIM_GREEN = '#6b9a76';
+  const GREY = '#38414e';
+  const DARKER_GREY = '#212a37';
+  const LIGHT_GREY = '#9ca5b3';
+  const LIGHT_BROWN = '#746855';
+  const DARK_BLUE = '#1f2835';
+  const BEIGE = '#f3d19c';
+  const GREY_BLUE = '#2f3948';
+  const BLACK_BLUE = '#17263c';
+  const DARK_BABYBLUE = '#515c6d';
```

Comment:
move these to the top of the file. right now, these variables will get instantiated every time initMap() because it is in the method's scope. if we move it to the top of the file, outside of the methods, these variables will only be instantiated once